geoips/interfaces/yaml_based/products.py

```diff
@@ -140,8 +138,9 @@ def get_plugin(self, source_name, name, product_spec_override=None):
     def get_plugins(self):
         """Retrieve a plugin by name."""
         plugins = []
-        for source_name, name in self._unvalidated_plugins.keys():
-            plugins.append(self.get_plugin(source_name, name))
+        for source_name in self._unvalidated_plugins[self.name].keys():
```

Comment:
I think we should create an issue to improve the efficiency of this function later on. It's fast for every interface besides 'Products'. Thinking about it, the way we implemented this PR, forces the brunt of the work on this function, especially for interfaces with numerous plugins. I'd recommend we have a conversation about this to figure out the best way to move forward on this.

Reply:
@evrose54 Did you get what you needed on this comment? Did you create an issue for it?

Reply:
Yup! This can be resolved. I'm going to do that now.